azext_edge/edge/providers/support/base.py

```diff
@@ -504,11 +450,54 @@ def _capture_pod_container_logs(
                         "data": log,
                         "zinfo": (
                             f"{pod_namespace}/{resource_api.moniker}"
-                            f"/pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
+                            f"/{child_group}pod.{pod_name}.{container.name}.{zinfo_previous_segment}log"
                         ),
                     }
                 )
             except ApiException as e:
                 logger.debug(e.body)
 
     return processed
+
+
+def _process_kubernetes_resources(
+    resources: object,
+    resource_api: EdgeResourceApi,
+    child_group: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+    kind: Optional[str] = None,
+) -> List[dict]:
+    processed = []
+
+    if not prefix_names:
+        prefix_names = []
+
+    logger.info(f"Detected {len(resources.items)} {kind}s.")
```

Comment:
kind is indicated Optional with default None. If no kind is passed in, this log doesn't make sense. Also later logic will assign kind to r.kind

Reply:
changed it to required